newsroom/upload.py

```diff
@@ -1,21 +1,36 @@
-
 import flask
 import newsroom
 import bson.errors
 
 from werkzeug.wsgi import wrap_file
+from werkzeug.http import parse_range_header
 from werkzeug.utils import secure_filename
 from flask import request, url_for, current_app as newsroom_app
 from superdesk.upload import upload_url as _upload_url
 from superdesk import get_resource_service
 from newsroom.decorator import login_required
 
-
 cache_for = 3600 * 24 * 7  # 7 days cache
 ASSETS_RESOURCE = 'upload'
 blueprint = flask.Blueprint(ASSETS_RESOURCE, __name__)
 
 
+class MediaFileLoader:
+    _loaded_files = {}
+
+    @classmethod
+    def get_media_file(cls, media_id):
+        if media_id in cls._loaded_files:
+            return cls._loaded_files[media_id]
```

Comment:
Seems like this will cache media files. This will increase memory usage of the app for each new asset requested.
@marwoodandrew Is this really required, as system resources would be consumed and an OOM scenario will be raised by the OS and start killing services to free RAM.